src/passes/RemoveNonJSOps.cpp

```diff
@@ -435,45 +301,1038 @@ struct RemoveNonJSOpsPass : public WalkerPass<PostWalker<RemoveNonJSOpsPass>> {
     Name functionCall;
     switch (curr->op) {
       case NearestFloat32:
-        needNearestF32 = true;
         functionCall = WASM_NEAREST_F32;
         break;
       case NearestFloat64:
-        needNearestF64 = true;
         functionCall = WASM_NEAREST_F64;
         break;
 
       case TruncFloat32:
-        needTruncF32 = true;
         functionCall = WASM_TRUNC_F32;
         break;
       case TruncFloat64:
-        needTruncF64 = true;
         functionCall = WASM_TRUNC_F64;
         break;
 
       case PopcntInt32:
-        needPopcntInt32 = true;
         functionCall = WASM_POPCNT32;
         break;
 
       case CtzInt32:
-        needCtzInt32 = true;
         functionCall = WASM_CTZ32;
         break;
 
       default: return;
     }
+    neededIntrinsics.insert(functionCall);
     replaceCurrent(builder->makeCall(functionCall, {curr->value}, curr->type));
   }
-
-private:
-  std::unique_ptr<Builder> builder;
 };
 
 Pass *createRemoveNonJSOpsPass() {
   return new RemoveNonJSOpsPass();
 }
 
+namespace {
+// A large WAST blob which contains the implementations of all the intrinsics
```

Comment:
the blob could be a separate `.wast` file which we load using the helpers in `wasm-io.h`. The only non-obvious thing is getting the file's location, I suppose, so maybe that's a bad idea.

Reply:
Another option is to `#include` it at compile time, which could almost work, but it wouldn't be quoted, hmm...

Reply:
Actually I think we *can* `#include` it at compile time, basically just
```
const char *IntrinsicsModule = R"(
#include "stuff.wast"
)";
```
Should be no quoting problem there... I think.

Reply:
Ah unfortunately I think that's just a string literal with `#include` in it :(

(didn't work locally)

Reply:
Hm it looks like somehow a different strategy will be required here, MSVC is barfing on a "too long" string literal in any case.

Reply:
This might be an option: https://stackoverflow.com/questions/410980/include-a-text-file-in-a-c-program-as-a-char/25021520#25021520 (other options in there too)

Reply:
Another option is to use an archive and `Linker::linkObject()` or `::linkArchive()` (see example use in `src/tools/s2wasm.cpp`). That's probably a more "proper" option, but it would mean that wasm2asm would need to be called with a parameter of where to find the file(s). If we expect more stuff to appear here, though, then maybe that's worth considering.

Reply:
Or maybe we could use `wasm-linker` that to link even if the file is embedded in the C++. That might be nicer/safer than manually copying the files ourselves, but I'm not sure. I think @dschuff and @jgravelle-google know that code best - thoughts?

Reply:
Ok I've moved this from a string literal to a file that's generated as part of the build using a giant array listed in a static. Apparently MSVC doesn't have the same limits on that! I believe CI should be green afterwards

Reply:
`wasm-linker` is very `s2wasm`-specific, and only "links" symbols internal to a module, i.e. it does data layout.

From `wasm-linker.h`:
> (In particular, there is no merging of multiple modules). Currently this is only inteded to turn a .s file produced by LLVM into a usable wast file.

I'd think `wasm-merge` would be what you'd want instead? My understanding  (and a quick glance at the code) is it mostly resolves imports/exports, which is what we want here.